### PR TITLE
feat(ProjectUI): Custom Field for Additional data in Proejct

### PR DESCRIPTION
**Pull request by akapti**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
* Enabled custom field for additional data in project portlet.

Issue: #860 

### How To Test?
* Please follow the steps mentioned in issue #809 for testing.

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

Signed-off-by: Abdul Kapti <abdul.mannankapti@siemens.com>

### CODE REVIEW

**Comment by JaideepPalit** [comment]
Created the Custom Fields as follows
`Control Panel -> Configuration->Custom Fields -> User -> add custom field` with `admin@sw360.org` - working fine
After logout and login as `user@sw360.org` -> Open Project - not loading, getting `Exception`

```
2020-05-28 12:24:09.297 ERROR [http-nio-8080-exec-51][render_portlet_jsp:131] null
java.lang.NullPointerException
	at org.eclipse.sw360.portal.common.CustomFieldHelper.lambda$null$0(CustomFieldHelper.java:146)
	at java.util.HashMap.forEach(HashMap.java:1289)
	at org.eclipse.sw360.portal.common.CustomFieldHelper.lambda$getCustomFields$1(CustomFieldHelper.java:123)
	at java.util.HashMap.forEach(HashMap.java:1289)
	at org.eclipse.sw360.portal.common.CustomFieldHelper.getCustomFields(CustomFieldHelper.java:98)
	at org.eclipse.sw360.portal.common.PortletUtils.setCustomFieldsDisplay(PortletUtils.java:435)
	at org.eclipse.sw360.portal.portlets.projects.ProjectPortlet.prepareDetailView(ProjectPortlet.java:1161)
	at org.eclipse.sw360.portal.portlets.projects.ProjectPortlet.doView(ProjectPortlet.java:1039)
```

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/PortletUtils.java [hunk 1]
```diff
@@ -431,6 +431,11 @@ public static void setWelcomePageGuideLine(RenderRequest request) {
         request.setAttribute("welcomePageGuideLine", welcomePageGuideLine);
     }
 
+    public static void setCustomFieldsDisplay(RenderRequest request, User user, Project project) {
```
[the remark this answers is not in the document]
Comment (akapti) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/PortletUtils.java:428]: I don't see any significance of using Generic method, as it requires additional effort of identifying the object type and `typecasting` the object to respective class (Project, Component or Release) in order to get the `additionalData` field. Kindly suggest if there is a way to achieve it using generic method but without `typecasting`.

**Comment by akapti** [comment]
> For Projects which has no value for custom field in DB. In UI appears like data present with empty value in additional data for the custom fields. which may vary when same Project record is fetched from REST.
* The above issue has been fixed

<br>

> Custom Field Type - Radio is not working.
* Expected behavior.

<br>

> After logout and login as user@sw360.org -> Open Project - not loading, getting Exception
* This has been fixed.

**Comment by JaideepPalit** [comment]
Tested above issues are working
- able to load project from USER Role - `working fine`
- if additional data is empty, not showing in UI. - `working fine`

Tested successfully, except -
For Dropdown , values are limited to `Option1`, `Option2` in UI , but it can be manipulated to different value like `Diff_Option` from REST - Update Project, same for checkbox.

**Pull request merged**